api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2294,6 +2297,182 @@ public void copyObject(String bucketName, String objectName, String destBucketNa
     }
   }
 
+
+  /**
+   * ComposeObject - It creates an object using server-side copying of
+   * existing objects. It takes a list of source objects and concatenates
+   * them into a new object using only server-side copying operations
+   *
+   * </p>
+   * <b>Example:</b><br>
+   *
+   * <pre>
+   * {@code minioClient.composeObject(String destinationBucketName, String destintionObjectName,
+   *                              List<CopySource> listSourceObjects);}
+   * </pre>
+   *
+   * @param destinationBucketName
+   *          Destination Bucket to be created upon compose.
+   * @param destintionObjectName
+   *          Destination Object to be created upon compose.
+   * @param listSourceObjects
+   *          List of Source Objects used to compose Object
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InsufficientDataException  upon getting EOFException while reading given
+   *           InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidArgumentException    upon invalid value is passed to a method.
+   */
+
+  public void  composeObject(String destinationBucketName, String destintionObjectName,
+                             List<CopySource> listSourceObjects)
```

Comment:
`copySources` is much better than `listSourceObjects`

Reply:
done.